### PR TITLE
ДЗ5

### PR DESCRIPTION
**Pull request by ggeorg0**
I asked ChatGPT to write a small poem of why sending home assignment fifteen minutes after deadline it is not not a big deal

> Fifteen minutes, just a breath,
> Not a crime, nor late by death.
> The clock’s stern face may softly chide,
> Yet wisdom knows, no need to hide.
> 
> Ideas bloom in time’s embrace,
> Rushing leaves the finer trace.
> A moment more, a final tweak,
> Perfection calls, the bold and meek.
> 
> So judge not harsh this fleeting span,
> A human heart, not just a plan.
> For fifteen minutes, barely seen,
> Are ripples in a calm routine.

### CODE REVIEW

File: project1.sql [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    user_id SERIAL PRIMARY KEY,
+    name VARCHAR(255) NOT NULL,
+    email VARCHAR(255) UNIQUE NOT NULL,
+    password_hash CHAR(128) NOT NULL,
+    created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+    is_active BOOLEAN DEFAULT TRUE
+);
+
+-- create type for admin permissions
+CREATE TYPE ADMIN_PERM_T AS ENUM ('read_only', 'dev');
```
Comment (dam-dscs) [inline project1.sql:11]: \+2 балла за создание собственного типа

File: project1.sql [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    name VARCHAR(255) NOT NULL,
+    email VARCHAR(255) UNIQUE NOT NULL,
+    admin_permissions ADMIN_PERM_T NOT NULL DEFAULT 'read_only',
+    is_super_admin BOOLEAN DEFAULT FALSE
+);
+
+CREATE TABLE prices (
+    price_id SERIAL PRIMARY KEY,
+        -- I found that PosgreSQL have a `MONEY` type
+    price_value MONEY NOT NULL CHECK (price_value > 0::MONEY),
```
Comment (dam-dscs) [inline project1.sql:24]: \+2 за проверку входящих данных

File: project1.sql [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+CREATE TABLE meetings (
+    meeting_id SERIAL PRIMARY KEY,
+    title VARCHAR(255) NOT NULL,
+    meet_desc TEXT,
+    meet_host INT REFERENCES users (user_id) ON DELETE SET NULL,
+    start_time TIMESTAMP NOT NULL,
+    end_time TIMESTAMP NOT NULL,
+    repeat_period PERIOD_T DEFAULT NULL,
+    location VARCHAR(511),
+    is_canceled BOOLEAN DEFAULT FALSE
```
Comment (dam-dscs) [inline project1.sql:58]: \+2 за значения по умолчанию

**Review by dam-dscs — APPROVED** [review]
Спасибо за работу, ставлю 10 базовых и 6 бонусных баллов

**Comment by ggeorg0** [comment]
Спасибо за проверку и комментарии!

**Pull request merged**